pkg/pillar/cmd/nim/controllerdns.go

```diff
@@ -70,28 +81,191 @@ func (n *nim) queryControllerDNS() {
 	}
 }
 
+func (n *nim) resolveWithSrcIP(domain string, dnsServerIP net.IP, srcIP net.IP) []dnsResponse {
```

Comment:
Seems this is a generic helper method for domain name resolution, not tight to nim (log could be an argument). Hence you could drop the receiver - also see my suggestions for `resolveWithPortsLambda` about moving to `devicenetwork` package.